### PR TITLE
Support for Marker with multiple values

### PR DESCRIPTION
**Pull request by rambangaru**

### CODE REVIEW

**Review by iivanou — CHANGES_REQUESTED** [review]
1. Tests are failed.
2. Provide a use case for suggested changes.

Thanks in advance.

**Comment by rambangaru** [comment]
>     1. Tests are failed.
> 
>     2. Provide a use case for suggested changes.
> 
> 
> Thanks in advance.

In PyTest, Markers supports multiple values and when multiple values are used,  the current code takes only on and ignore the remaining.

Lets take a use case, Consider I'm using the plugin `https://github.com/Projectplace/pytest-tags` for the purpose of adding tags to my Tests and Test have multiple tags. PyTest RP Agent takes the very first tag and add to report portal portal and ignores remaining, this particular code change, supports to update all the mentioned tags, and we can expect something like below in report portal

image

**Comment by iivanou** [comment]
> > ```
> > 1. Tests are failed.
> > 
> > 2. Provide a use case for suggested changes.
> > ```
> > 
> > 
> > Thanks in advance.
> 
> In PyTest, Markers supports multiple values and when multiple values are used, the current code takes only on and ignore the remaining.
> 
> Lets take a use case, Consider I'm using the plugin `https://github.com/Projectplace/pytest-tags` for the purpose of adding tags to my Tests and Test have multiple tags. PyTest RP Agent takes the very first tag and add to report portal portal and ignores remaining, this particular code change, supports to update all the mentioned tags, and we can expect something like below in report portal
> 
> image
> 
> the current behavior is, it takes only first tag and report that alone to report portal
> 
> image

Ok. can you add unit tests then?

**Comment by rambangaru** [comment]
> > > ```
> > > 1. Tests are failed.
> > > 
> > > 2. Provide a use case for suggested changes.
> > > ```
> > > 
> > > 
> > > Thanks in advance.
> > 
> > 
> > In PyTest, Markers supports multiple values and when multiple values are used, the current code takes only on and ignore the remaining.
> > Lets take a use case, Consider I'm using the plugin `https://github.com/Projectplace/pytest-tags` for the purpose of adding tags to my Tests and Test have multiple tags. PyTest RP Agent takes the very first tag and add to report portal portal and ignores remaining, this particular code change, supports to update all the mentioned tags, and we can expect something like below in report portal
> > image
> > the current behavior is, it takes only first tag and report that alone to report portal
> > image
> 
> Ok. can you add unit tests then?

Added unit test. I actually added additional test data for existing test to validate new addition.

Kindly review.

File: tests/test_service.py [hunk 1]
```diff
@@ -37,6 +41,8 @@ def __iter__(self):
     assert markers == [{'value': 'test_marker'},
                        {'key': 'test_decorator_key',
                         'value': 'test_decorator_value'},
+                       {'key': 'test_decorator_key_with_multi_value',
```
Comment (iivanou) [inline tests/test_service.py:44]: In this case, your marker will be displayed at RP as: `test_decorator_key_with_multi_value:['test_value1', 'test_value2']`. I do not think that is what you wanted to achieve...

Comment (rambangaru) [reply tests/test_service.py:43]: my apologies, I was curious and testing how it react if I pass list and accidentally pushed. 

Fixed tests and it should be good now.

`{'key': 'test_decorator_key_with_multi_value',
  'value': 'test_value1'},
  {'key': 'test_decorator_key_with_multi_value',
  'value': 'test_value2'},`

**Review by iivanou — APPROVED** [review]

**Pull request merged**